### PR TITLE
feat: add possibility to specify customized bonus points logic for live quizzes

### PR DESCRIPTION
**Pull request by sjschlapbach**
Screenshot 2024-09-17 at 15 18 09
Screenshot 2024-09-17 at 13 15 42

### CODE REVIEW

**Review by rschlaefli — APPROVED** [review]

File: packages/prisma/src/prisma/migrations/20240917085550_modular_live_quiz_bonus_points/migration.sql [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+ALTER TABLE "LiveQuiz" ADD COLUMN     "maxBonusPoints" INTEGER NOT NULL DEFAULT 45,
+ADD COLUMN     "timeToZeroBonus" INTEGER NOT NULL DEFAULT 20;
+
+-- AlterTable
+ALTER TABLE "LiveSession" ADD COLUMN     "maxBonusPoints" INTEGER NOT NULL DEFAULT 45,
+ADD COLUMN     "timeToZeroBonus" INTEGER NOT NULL DEFAULT 20;
+
+-- AlterTable
+ALTER TABLE "QuestionInstance" ADD COLUMN     "maxBonusPoints" INTEGER,
+ADD COLUMN     "timeToZeroBonus" INTEGER;
```
[the remark this answers is not in the document]
Comment (sjschlapbach) [reply packages/prisma/src/prisma/migrations/20240917085550_modular_live_quiz_bonus_points/migration.sql:11]: The values on the instance are allowed to be null, the corresponding value is set to a default value, if not set in the grading logic. For all future instances, this value should automatically be set. However, this is only the case for live quizzes and practice quizzes / microlearnings, using the same element instance as live quizzes in the future will never have this parameter set

File: packages/i18n/index.ts [hunk 2]
```diff
@@ -23,6 +23,6 @@ export function getMessageFallback({
   if (error.code === IntlErrorCode.MISSING_MESSAGE) {
     return `${path} is not yet translated`
   } else {
-    return `Dear developer, please fix this message: ${path}`
+    return `Dear developer, please fix this message: ${path}. Error: ${error.message}`
```
[the remark this answers is not in the document]
Comment (sjschlapbach) [reply packages/i18n/index.ts:26]: This message is only meant for development

File: packages/graphql/src/graphql/ops/MCreateSession.graphql [hunk 3]
```diff
@@ -5,6 +5,8 @@ mutation CreateSession(
   $blocks: [BlockInput!]!
   $courseId: String
   $multiplier: Int!
+  $maxBonusPoints: Int!
+  $timeToZeroBonus: Int!
```
[the remark this answers is not in the document]
Comment (sjschlapbach) [reply packages/graphql/src/graphql/ops/MCreateSession.graphql:9]: Not required, mutation calls are updated as well

File: apps/frontend-manage/src/components/sessions/creation/liveQuiz/AdvancedLiveQuizSettings.tsx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+              <RechartsTooltip
+                content={({ active, payload }) => {
+                  if (active && payload && payload.length) {
+                    const time = payload[0].payload.time
+                    const points = payload[0].payload.points
+
+                    return (
+                      <div className="border-primary-100 rounded border border-solid bg-white p-2 text-gray-600">
+                        <div className="font-bold">
+                          {payload[0].payload.windowStart}
```
[the remark this answers is not in the document]
Comment (sjschlapbach) [reply apps/frontend-manage/src/components/sessions/creation/liveQuiz/AdvancedLiveQuizSettings.tsx]: removed

**Pull request merged**